IHP/AuthSupport/Authorization.hs

```diff
@@ -5,21 +5,21 @@ Copyright: (c) digitally induced GmbH, 2020
 -}
 module IHP.AuthSupport.Authorization where
 
-import IHP.Prelude
+import IHP.Controller.Render
```

Comment:
@mpscholten I might need your help in untangling this part which creates a cyclic import. I couldn't find an easy way 

![image](https://github.com/digitallyinduced/ihp/assets/125707/cf9fc1e4-e5f6-4dcb-9b22-87da16ce18b1)

Another option is to move `accessDeniedUnless` into `IHP.ErrorController`

Reply:
What do you think about moving all the functions related to 403 responses (including `accessDeniedUnless`) to it's own `IHP.Controller.AccessDenied` module?

Reply:
> What do you think about moving all the functions related to 403 responses

I'm not sure.

1. IMO we should keep `handleAccessDeniedFound` in `IHP/ErrorController.hs` next to the `handleNotFound` - they are very similar, and maybe we can even create helper functions to reduce them.
2. `renderNotFound` and `renderAccessDenied` also feel natural inside `IHP/Controller/Render.hs`

So I think the question is only about `accessDeniedUnless` and `accessDeniedWhen`.  Why not inside `IHP.ErrorController`?

Reply:
The `IHP.ErrorController` should only have code related to actually error handling. `accesDeniedUnless` is not directly related to that.

Maybe we can move `accessDeniedUnless` to `IHP.ControllerSupport`?

Reply:
Yeah, I like that - let me try it out.

Another concern I have. We have this code that I think caches the the 404 pages?

https://github.com/digitallyinduced/ihp/blob/45902cce813a378f413f05628defec2e1e493a7d/IHP/Server.hs#L100

But there's no `Static.ss403Handler`. Should we create a PR for [Network.Wai.Application.Static](https://hackage.haskell.org/package/wai-app-static-3.1.7.4/docs/Network-Wai-Application-Static.html)?

Reply:
The `ss404Handler` is called whenever a file is not found in `static/` that should match the request.

With latest master IHP routing prio is:
- app routes (e.g. controllers and actions)
- files that exist in `static/`
- files that exists in `IHP/static`
- when nothing matches, render 404 (via `ss404Handler`)

Reply:
> Maybe we can move accessDeniedUnless to IHP.ControllerSupport?

I think the place is right, but still got a cyclic import. I don't have a clear mental model of who should be importing what.

Reply:
@mpscholten How about `IHP.Controller.Render`? As it's really just a render function in the end

Reply:
Hm, doesn't feel like it belongs there. I think this might get tricky when we don't want to move it into it's own dedicated module.

If we end up with something like `IHP.Controller.AccessDenied` we should also consider doing `IHP.Controller.NotFound` btw. The `ErrorController` module is already quite large, so it would be good to split up (this helps with compile times as well)

Reply:
Ok, I'll try so separate to `IHP.Controller.AccessDenied` and `IHP.Controller.NotFound`

Reply:
I'll also add `notFoundWhen`. Sometimes one would like to return 404 instead of 403 -- to prevent sniffing URLs and knowing they hit the right one by getting a 403.